watertap/costing/units/rectifier.py

```diff
@@ -0,0 +1,89 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
```

Comment:
What was the key takeaway point on why we'd want to make this a separate "unit" instead of keeping this in costing/util? I saw a lot of back and forth on Slack, but it would be good to get a clear summary.

Reply:
So the current version [0b0e7ec](https://github.com/watertap-org/watertap/pull/1069/commits/0b0e7ec94e390f6e45c815da7e76a919c028e5a0) is structured the way it is because it was the minimal amount of refactoring to not have this as the bottleneck for the ED analysis. 

The main discussion that then will have subsequent challenges to address in either implementation is:
1. Should the rectifier `captial_cost` exist on its own unit model costing block? If it exists on its own unit model costing block, it must also have a registered unit model block.
2.  If it exists as a component of electrodialysis.costing.captial_cost = (electrodialysis.costing.electrodialysis_capital_cost +  electrodialysis.costing.rectifier_capital_cost), is it a clear cost breakdown? Say two electrified unit ops are operating in parallel, supplied by the same rectifier. This costing method implies one half of the rectifier is being costed on each electrified unit costing block.